Assets/Scripts/Tags/Hazard.cs

```diff
@@ -1,45 +1,41 @@
 using UnityEngine;
+using UnityEngine.Serialization;
 
 public class Hazard : MonoBehaviour {
+    [Tooltip("Defines how high and low this object will bob up and down, measured in world-space units.")]
+    [SerializeField] private float amplitude = 0.2f;
 
-    [SerializeField] private float verticalSpeed;
-    [SerializeField] private float amplitude;
+    [Tooltip("The rate at which this object will rotate over time, measured in degrees per second.")]
+    [FormerlySerializedAs("rotSpeed")]
+    [SerializeField] private float angularSpeed = 60;
 
-    [SerializeField] private float rotSpeed;
+    [Tooltip("Defines the range within which a vertical speed will be randomly chosen for this object.\n\n" +
+        "This object's vertical speed describes the frequency of the up and down bobbing motion.")]
+    [SerializeField] private Vector2 verticalSpeedRange = new Vector2(2, 3);
 
-    private Vector3 origPos;
-    private Vector3 tempPos;
-
-    private Vector3 rot = new Vector3(45f, 45f, 45f);
-
-    private float rotDegreesPerSecond = 45f;
-
-    private static System.Random rndFloat = new System.Random();
-    private static System.Random rndRot = new System.Random();
```

Comment:
You are totally fine to use System.Random in C#, but you don't have to in Unity.
System.Random requires that you instantiate it (aka, use `new Random()`).

Also you can just add `using System;` to the top of your script, so you could write something like this for example:

```cs
using System;

//Further down in your script...
Random generator = new Random();
float value = (float) generator.NextDouble();
```

However, with Unity, you can use the [`UnityEngine.Random`](https://docs.unity3d.com/ScriptReference/Random.html) class, which is accessed statically instead.
Thus, you can just call:

```cs
using UnityEngine;

//Further down in your script...
int intValue = Random.Range(0, 5); //Returns an INTEGER in range [0, 4], which is equivalent to [0, 5).
float floatValue = Random.Range(0f, 5f) //Returns a FLOAT in range [0.0, 5.0], including any decimal numbers in between.
```

Reply:
I understand now. Yes, there were two Random classes I could use so I like your final suggestion. Much cleaner.